From e87262ee309a1bbd250f4cfb04cde34112589f3f Mon Sep 17 00:00:00 2001
From: ravindranath <ravindranathx.doddi@intel.com>
Date: Fri, 8 Feb 2013 18:55:21 -0500
Subject: [PATCH 2/2] Add libusb support

libusb support is required for userspace usb driver in Bluedroid stack.
Build is fixed for android in io.c by including macro TIMESPEC_TO_TIMEVAL

Issue: AXIA-1459
Change-Id: I7d3480b9e7aee45e4be0d82d7e65d3b914a615d8
Signed-off-by: Ravindranath Doddi <ravindranathx.doddi@intel.com>
---
 config.h          | 13 +++++++++----
 libusb/Android.mk | 20 ++++++++++++++++++++
 libusb/io.c       |  5 +++++
 3 files changed, 34 insertions(+), 4 deletions(-)
 create mode 100644 libusb/Android.mk

diff --git a/config.h b/config.h
index 9007586..e251d64 100644
--- a/config.h
+++ b/config.h
@@ -1,6 +1,4 @@
-/* config.h.  Generated from config.h.in by configure.  */
-/* config.h.in.  Generated from configure.ac by autoheader.  */
-/* saintlou: File is identical with PPC and Intel, 10/26/2010 */
+/* config.h. is modified to support bluedroid */
 
 /* Default visibility */
 #define API_EXPORTED __attribute__((visibility("default")))
@@ -18,7 +16,9 @@
 #define HAVE_INTTYPES_H 1
 
 /* Define to 1 if you have the `rt' library (-lrt). */
-/* #undef HAVE_LIBRT */
+#ifdef _SHARED_LIBRARY_
+#define HAVE_LIBRT 1
+#endif
 
 /* Define to 1 if you have the <memory.h> header file. */
 #define HAVE_MEMORY_H 1
@@ -84,10 +84,15 @@
 #define STDC_HEADERS 1
 
 /* Backend handles timeout */
+#ifndef _SHARED_LIBRARY_
 #define USBI_OS_HANDLES_TIMEOUT /**/
 
 /* timerfd headers available */
 /* #undef USBI_TIMERFD_AVAILABLE */
+#else
+#undef USBI_OS_HANDLES_TIMEOUT
+#undef USBI_TIMERFD_AVAILABLE
+#endif
 
 /* Version number of package */
 #define VERSION "1.0.8"
diff --git a/libusb/Android.mk b/libusb/Android.mk
new file mode 100644
index 0000000..c437a87
--- /dev/null
+++ b/libusb/Android.mk
@@ -0,0 +1,20 @@
+LOCAL_PATH:= $(call my-dir)
+
+include $(CLEAR_VARS)
+
+LOCAL_SRC_FILES:= \
+ core.c \
+ descriptor.c \
+ io.c \
+ sync.c \
+ os/linux_usbfs.c
+
+LOCAL_C_INCLUDES += \
+ external/libusb/ \
+ external/libusb/libusb/ \
+ external/libusb/libusb/os
+
+LOCAL_CFLAGS := -D_SHARED_LIBRARY_
+LOCAL_MODULE_TAGS:= optional
+LOCAL_MODULE:= libusb
+include $(BUILD_SHARED_LIBRARY)
diff --git a/libusb/io.c b/libusb/io.c
index 1379eb3..132721c 100644
--- a/libusb/io.c
+++ b/libusb/io.c
@@ -36,6 +36,11 @@
 
 #include "libusbi.h"
 
+#define TIMESPEC_TO_TIMEVAL(tv, ts)                                     \
+        do {                                                            \
+                (tv)->tv_sec = (ts)->tv_sec;                            \
+                (tv)->tv_usec = (ts)->tv_nsec / 1000;                   \
+        } while (0)
 /**
  * \page io Synchronous and asynchronous device I/O
  *
-- 
1.9.1

